pandas/_libs/tslibs/conversion.pyx

```diff
@@ -445,7 +448,7 @@ cpdef int64_t tz_convert_single(int64_t val, object tz1, object tz2):
     if get_timezone(tz2) == 'UTC':
         return utc_date
     if is_tzlocal(tz2):
-        dt64_to_dtstruct(val, &dts)
+        dt64_to_dtstruct(utc_date, &dts)
```

Comment:
huh? this seems like it would break something, if it doesn't that's a problem

Reply:
The bug here is an inconsistency between tz_convert and tz_convert_single. The edit here makes tz_convert_single behave the same as tz_convert.

The reason why this hasn't broken anything is because in all existing usages+tests of tz_convert_single, one of `tz1` or `tz2` is UTC.  As long as this condition holds, the two versions of the code are equivalent.  This PR adds an assertion that this condition holds.

Of course, _assuming the tz_convert imlementation is correct_, once the implementation here is changed, the assertion is no longer necessary.

Reply:
I think one of my comments is gone. remove the first clause else (which just sets ``utc_date = val``, then this is no longer necessary.